packages/opentelemetry-exporter-collector-grpc/src/util.ts

```diff
@@ -105,6 +105,13 @@ export function send<ExportItem, ServiceRequest>(
   }
 }
 
-function removeProtocol(url: string): string {
-  return url.replace(/^https?:\/\//, '');
+export function fixUrl(url: string): string {
```

Comment:
`validateAndNormalizeUrl` sound good?

Reply:
It still is kind of the same, fix, validate etc., I would not guess what it really does. What about `getHostFromUrl` which is in fact exactly what it does ?

Reply:
because it also validates that the protocol either does not exist, or is a supported protocol; likewise same with the path.

Reply:
Validation/normalization rules can change so the name shouldn't be too specific to the implementation anyway. For example we might use proto for something (port selection) in the future.

Reply:
I don't like the name but will not block it because of the name :). 
But there is something else here which might be more serious. Previously setting the wrong url was safe in the sense that it would not break the app. Now having a wrong url for the exporter might raise the validation error directly from `URL` for example
```
new URL('123')
// error raised, app broken ?
```
I don't think it should work like this, we have rule we should not break user app and here it will break. The exporter might now work, but the app should be be usable.

Reply:
There was discussion of this in another thread, and the decision from TC was that it was acceptable (and encouraged in the spec) to hard-fail early in initialization; what's not allowed is to fail once the app has started running.

https://github.com/open-telemetry/opentelemetry-js/pull/2130#discussion_r617813643

Reply:
sounds fine for me then

Reply:
resolving this thread, continuing discussion above, mark it resolved when you're satisfied or not :)